File: ECOMEP.Angular/src/app/shared/models/time-entry-dto.ts
import { Contact } from "src/app/contact/models/contact";
import { BaseEntity } from "src/app/shared/models/base-entity.model";

export class TimeEntryDto extends BaseEntity
{
  contactID: number = 0;
  contact: Contact = new Contact();
  taskTitle!: string;
  manHours: number = 0;
  wFTaskID?: number = 0;
  isPaused: boolean = false;
  projectID?: number = 0;
  companyID: number = 0;
  manValue: number = 0;
  valueHourRate: number = 0;
  valueHourCost: number = 0;
  startDate: Date = new Date();
  endDate?: Date;
  wfTaskID: number = 0;
  mHr: number = 0;
  vHr: number = 0;

  constructor(init?: Partial<BaseEntity>)
  {
    super(init);
    Object.assign(this, init);

  }
}

File: ECOMEP.WebApi/MyCockpitView.WebApi/WFTaskModule/Dtos/TimeEntryDto.cs
using MyCockpitView.WebApi.ContactModule.Dtos;
using MyCockpitView.CoreModule;



namespace MyCockpitView.WebApi.WFTaskModule.Dtos;

public class TimeEntryDto : BaseEntityDto
{

    public int ContactID { get; set; }
    public virtual ContactListDto? Contact { get; set; }


    public string? TaskTitle { get; set; }

    public decimal ManHours { get; set; }


    public DateTime StartDate { get; set; }


    public DateTime? EndDate { get; set; }

    public int? WFTaskID { get; set; }

    public bool IsPaused { get; set; }

    public int? ProjectID { get; set; }
    public int CompanyID { get; set; }
    public decimal ManValue { get; set; }
    public decimal ValueHourRate { get; set; } = 0;
    public decimal ValueHourCost { get; set; } = 0;
    public int? EntityID { get; set; }
    public string? Entity { get; set; }
    public string? EntityTitle { get; set; }
}